language/move-prover/tests/sources/arithm.move

```diff
@@ -0,0 +1,71 @@
+module TestArithmetic {
+
+    // Most of the tests here just ensure that what the bytecode operation does is the same as the
+    // spec expressions.
+
+	fun add_two_number(x: u64, y: u64): (u64, u64) {
+		let res: u64 = x + y;
+		let z: u64 = 3;
+		(z, res)
+	}
+	spec fun add_two_number {
+	    aborts_if x + y > 9223372036854775807;
+	    ensures result_1 == 3;
+	    ensures result_2 == x + y;
+	}
+
+	fun multiple_ops(x: u64, y: u64, z: u64): u64 {
+		x + y * z
+	}
+	spec fun multiple_ops {
+        ensures result == x + y * z;
+    }
+
+	fun bool_ops(a: u64, b: u64): (bool, bool) {
+        let c: bool;
+        let d: bool;
+        c = a > b && a >= b;
+        d = a < b || a <= b;
+        if (!(c != d)) abort 42;
+        (c, d)
+    }
+    spec fun bool_ops {
+        ensures result_1 == (a > b && a >= b);
```

Comment:
I recall that we used to have result.0 and result.1.  We are now numbering starting from 1 and using _ instead of .  Any particular reason?

Reply:
The move parser does not support this syntax (before btw we had RET(1)). We would need to extend it for this, but I feel its not worth doing this. We should keep derivation from move syntax small.

Reply:
result_1 is probably not a reserved word.  Users who choose that name for a variable are probably going to have trouble.
No biggy for now.

Reply:
We probably need to extend the parser to support something special here.

Reply:
But I do think we should go with 0-based offsets.